### ISSUE TITLE
why divide the lase

### ISSUE THREAD

**Issue by ymq1117** [reiniscimurs/GDAM#12]
![image](https://github.com/reiniscimurs/GDAE/assets/59954416/ed316bc4-1699-41d6-8869-753d2df8747a)
Here, after dividing the readings into 19 groups, why divide the laser state by 10? If I still use simulated laser in Project drl-robot-navigation to divide 360 readings into 20 groups, do I still need to divide by 10?

**Comment by reiniscimurs** [comment]
Hi,
We normalize the laser value in range [0, 1].
First, we cap each laser value at 10 meters, then we divide it into groups, then we normalize the range. So even with more laser values, you would divide it by 10 to normalize the values. 

The value would have to be the same as here:
https://github.com/reiniscimurs/GDAE/blob/fc793eda8de23bed98ba3acd32908843c535510f/Code/GDAM_env.py#L188-L189

**Comment by ymq1117** [comment]
Thanks for your time and patience.
When I launch slam_toolbox by running "roslaunch slam_toolbox online_sync.launch", I want it to subscribe to "r1/front_laser/scan“  like the picture below. (The picture is from another issue)
![ab13c7bc74e52e079c840d4dcf0004b](https://github.com/reiniscimurs/GDAE/assets/59954416/344343db-21ea-4671-a7b5-94a7acd6a5d2)
But it subscribes to Topic "/scan" by default. I searched online but couldn't find where should I change it.
Could you tell me where to modify?

**Comment by reiniscimurs** [comment]
Hi,

have you updated your slam_toolbox configs with the topic names that you want to use?
For example:
https://github.com/SteveMacenski/slam_toolbox/blob/ros2/config/mapper_params_online_sync.yaml#L16

You would have to modify this for the method of SLAM toolbox that you are using.

**Comment by ymq1117** [comment]
Thanks so much for your time and patience!
After I launch move_base and slam_toolbox and run GADM.py， here is the output: This warning keeps coming up
![5e04103f00f9a7d921e38329cd59258](https://github.com/reiniscimurs/GDAE/assets/59954416/8bf0f832-10f1-4e77-997f-840d292e9a21)
And this error appears after I execute the command "Ctrl+c".
![dfdc0be3638e1f75da01cd0e33acabb](https://github.com/reiniscimurs/GDAE/assets/59954416/863596dc-bb31-4b04-8f52-785cac526413)
tf2.LookupException: "map" passed to lookupTransform argument target_frame does not exist. 
Do you have any idea?

**Comment by reiniscimurs** [comment]
I do not recall where the map would be set and where the connection would be made. Perhaps in your costmap configs.

You could check if you have connection between your map topic and base_link in the qrt.

Additionally you could try publishing connection directly in your launch file and seeing if that works:


```
<node pkg="tf" type="static_transform_publisher" name="map_to_odom"
args="0.0 -0.0 0.0 3.14 0.0 0.0 map base_link  200" />

```

**Comment by ymq1117** [comment]
Thank you for your help and patience ! 
I'm sorry I'm not familiar with ros. This is my launch file and yaml configuration file for move_base and the topics I modified in GDAM_env.py, and also the laser input. These are all the modifications I've done to the original project.
 There are still many problems with the output.
![c425f230f6116ac8f467956949667c2](https://github.com/reiniscimurs/GDAE/assets/59954416/6f0684a7-2664-4575-ad56-b983ec5daa53)

![bf29da8631d78c06296be276f57b923](https://github.com/reiniscimurs/GDAE/assets/59954416/f211bf15-a72e-4b92-8361-2b23d82f9a6c)
![1cc33ae9a6e03eb081447b53998eea1](https://github.com/reiniscimurs/GDAE/assets/59954416/9175a276-8636-4b0a-8803-8c3ee1919bdb)
![d5cf7d4dfeeeacf9dccfe26e6ef2a98](https://github.com/reiniscimurs/GDAE/assets/59954416/999bd3d7-3cd4-4f55-967f-d2e8fa20d790)
![6625484220917ffbc046f13fcabcf0b](https://github.com/reiniscimurs/GDAE/assets/59954416/2ff05b91-ef78-4639-bd3b-b4ed5860ebf3)
![fbe579d1a912bde1419e557c646084e](https://github.com/reiniscimurs/GDAE/assets/59954416/8a3e67c5-8482-45c4-855a-4452dffee6bc)
![250d31e760e0c7179f6f22fba3221c5](https://github.com/reiniscimurs/GDAE/assets/59954416/87cfb4ab-d6af-4877-964b-f8324e5b147e)

![graph](https://github.com/reiniscimurs/GDAE/assets/59954416/43c31498-fd66-4959-a1a2-6fd5ea2cfeb9)
![frame](https://github.com/reiniscimurs/GDAE/assets/59954416/1c6e3370-20e3-4622-9077-030a7685809a)
![1697981488874](https://github.com/reiniscimurs/GDAE/assets/59954416/cdf2f1e8-2f44-4f6e-9bb7-681c822f08be)
![1697981549441](https://github.com/reiniscimurs/GDAE/assets/59954416/a004c548-88c1-452c-a283-3ecc43fe4b72)
![271d185a9af8958c7897a91029bebf5](https://github.com/reiniscimurs/GDAE/assets/59954416/259071a8-47e4-4cbf-a60e-ef644b20412d)
 Could your give me some suggestion to run it correctly? The "global frame" in the move_base configuration yaml files should be odom or map?

**Comment by reiniscimurs** [comment]
Hi, so I see a couple of things here.

You probably want to use `map` as `global_frame` in your costmaps. You currently do not have a link from `odom` to `base_link` (hence the error). See if using map help there. Then probably publishing it manually should not be required.

I am a little confused about where do the laser readings come from? And what are the topic names of your sensors? I see there is a velodyne lidar in the image, but that is not part of sensor suite in the repository. Are you merging the DRL repository with this one? In any case, you should check the topic names that you are trying to launch. For instance, I see you are trying to subscribe to `r1/odom` topic, but in the rqt tree it is `/odom` topic.

**Comment by ymq1117** [comment]
Yes I'm merging the DRL repository. The topics are the same as in DRL repository.
![image](https://github.com/reiniscimurs/GDAE/assets/59954416/f32df51f-ec3a-4712-ad21-0cf512ee34e2)
Does this need to change? For example, change /odom to /r1/odom

**Comment by ymq1117** [comment]
Yes I'm merging  DRL repository with this one.
 It seems that there is no need to manually publish the map to base link transformation, after removing "<node pkg="tf" type="static_transform_publisher" name="map_to_odom"
args="0.0 -0.0 0.0 3.14 0.0 0.0 map base_link  200" />", my tf tree looks right now:
![534e81c82c486aa15c7c570475d2b0b](https://github.com/reiniscimurs/GDAE/assets/59954416/0142bdb1-78f3-40f7-ab64-b707e6535ce4).RobotModel status in rviz is also no error. Now, are there any errors with this tf tree? If the "odom" in the tree should be "/r1/odom", where should I change it?

If I set the global_frame in global costmap.yaml here as "odom“:
![image](https://github.com/reiniscimurs/GDAE/assets/59954416/80c8f91c-5f78-4c43-a9da-60b60ffabf6e)
The output is like this:
![9526bcad81a97ed7b6211e47e75e9ec](https://github.com/reiniscimurs/GDAE/assets/59954416/0080b029-2e9b-4aa9-bd6d-bf644087c820)
If I set it as "map",
The output is same as before  after I execute the command "Ctrl+c". Like this:
![image](https://github.com/reiniscimurs/GDAE/assets/59954416/ab8816eb-ef4e-4f59-86d1-052b6c2a037c)

**Comment by reiniscimurs** [comment]
In the DRL training scenario we spawn a virtual robot with namespace `r1`. This is not the case in this repo, as an actual real robot was used, so no robot was spawned and no namespace was used. I think you will have to align the naming of the topics for the program to understand what is happening.

You could try looking at `rostopic list` at runtime and try to see what topics are published and at what names.
